src/test/kotlin/com/autonomousapps/model/AttrRefTest.kt

```diff
@@ -0,0 +1,14 @@
+package com.autonomousapps.model
+
+import com.autonomousapps.model.AndroidResSource.AttrRef.Companion.from
+import org.junit.jupiter.api.Assertions.assertNull
+import org.junit.jupiter.api.Test
+
+class AttrRefTest {
+
+    @Test
+    fun `question mark character as inlined text is not parsed as an AttrRef`() {
+        assertNull(from("android:text" to "?"))
```

Comment:
What do you think the result should be if the test were
```kotlin
assertNull(from("android:text" to "?foo"))
```
?

With your current changeset, the test fails because `from(...)` returns
```
expected: <null> but was: <AttrRef(type=attr, id=foo)>
Expected :null
Actual   :AttrRef(type=attr, id=foo)
```
which is probably not what we want? Presumably `android:text` should be able to display any valid string, including things of the form `"?foo"`, without the Android system thinking that it's a reference to some attribute named "foo".

Reply:
Unfortunately this is not how Android (AAPT) handles this special case 😖

For instance, this layout is invalid:

```xml
<TextView android:text="?test" />
```

and produces:

```
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":".\\app\\build\\intermediates\\incremental\\debug\\mergeDebugResources\\stripped.dir\\layout\\activity_main.xml","position":{"startLine":1}}],"original":"ERROR:.\\app\\build\\intermediates\\incremental\\debug\\mergeDebugResources\\stripped.dir\\layout\\activity_main.xml:1: AAPT: error: resource attr/test (aka com.example:attr/test) not found.\n    ","tool":"AAPT"}

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugResources'.
> A failure occurred while executing com.android.build.gradle.internal.res.LinkApplicationAndroidResourcesTask$TaskAction
   > Android resource linking failed
     ERROR:.\app\build\intermediates\incremental\debug\mergeDebugResources\stripped.dir\layout\activity_main.xml:1: AAPT: error: resource attr/test (aka com.example:attr/test) not found.
```

The same thing happen with resource files:

```xml
<resources>
    <string name="referrer_test">?test</string>
</resources>
```

produces:

```
AGPBI: {"kind":"error","text":"Android resource linking failed","sources":[{"file":".\\app\\src\\main\\res\\values\\styles.xml","position":{"startLine":8,"startColumn":4,"startOffset":324,"endLine":15,"endColumn":12,"endOffset":792}}],"original":"ERROR:.\\app\\src\\main\\res\\values\\styles.xml:9:5-16:13: AAPT: error: resource attr/test (aka com.example:attr/test) not found.\n    ","tool":"AAPT"}

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugResources'.
> A failure occurred while executing com.android.build.gradle.internal.res.LinkApplicationAndroidResourcesTask$TaskAction
   > Android resource linking failed
     ERROR:.\app\src\main\res\values\styles.xml:9:5-16:13: AAPT: error: resource attr/test (aka com.example:attr/test) not found.
```

So I guest the following test matches the expected Android behavior:

```kotlin
assertEquals(
  AttrRef(type = "attr", id = "foo"),
  from("android:text" to "?foo"),
)
```

---

But this bring one last special case. If someone were to use something like that:

```xml
<TextView android:text="\?test" />
```

Then, AAPT "sees" the string "?test", and the corresponding test should be:

```kotlin
assertNull(from("android:text" to "\\?foo"))
```

Which means we should not use `Regex.containsMatchIn(CharSequence)` but rather `Regex.matchEntire(CharSequence)`.
Then, the current `ATTR_REGEX.matchEntire(id) != null -> AttrRef(...)` would behave like AAPT.

Reply:
omg I hate AAPT.